planner/core/exhaust_physical_plans.go

```diff
@@ -399,12 +399,27 @@ func (p *LogicalJoin) constructIndexMergeJoin(
 		if hasPrefixCol {
 			continue
 		}
+		// If join keys is not prefix of inner index. The order of merge join can't be granted.
+		// So index merge join can't be constructed in this case
+		idxOff2KeyOff := make([]int, len(join.OuterJoinKeys))
```

Comment:
I think you can directly pass `helper.idxOff2KeyOff` to `constructIndexMergeJoin`?

Reply:
We can't do this or in other words. It's hard to do this. The `constructIndexMergeJoin` is also used for "inner is `TableScan`" without `helper`. And for other sides, the `idxOff2KeyOff` is not consistent with `helper.idxOff2KeyOff`. It doesn't have the value `-1`.